File: Kapsch.MobileApps/KapschMobile/shared/src/main/java/za/co/kapsch/shared/Models/PaginationListModel.java
package za.co.kapsch.shared.Models;

/**
 * Created by CSenekal on 2017/07/03.
 */

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PaginationListModel<T> {

    @SerializedName("HasPreviousPage")
    private boolean mHasPreviousPage;
    @SerializedName("HasNextPage")
    private boolean mHasNextPage;
    @SerializedName("PageIndex")
    private int mPageIndex;
    @SerializedName("PageSize")
    private int mPageSize;
    @SerializedName("TotalPages")
    private int mTotalPages;
    @SerializedName("TotalCount")
    private int mTotalCount;
    @SerializedName("Models")
    private List<T> mModels;

    public boolean getHasPreviousPage() {
        return mHasPreviousPage;
    }

    public boolean getHasNextPage() {
        return mHasNextPage;
    }

    public int getPageIndex() {
        return mPageIndex;
    }

    public int getPageSize() {
        return mPageSize;
    }

    public int getTotalPages() {
        return mTotalPages;
    }

    public int getTotalCount() {
        return mTotalCount;
    }

    public List<T> getModels() {
        return mModels;
    }
}

File: Kapsch.Gateway.Models.Shared/PaginationListModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kapsch.Gateway.Models.Shared
{
    public class PaginationListModel<T>
    {
        public bool HasPreviousPage
        {
            get
            {
                return (PageIndex > 0);
            }
        }

        public bool HasNextPage
        {
            get
            {
                return (PageIndex + 1 < TotalPages);
            }
        }

        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int TotalPages { get { return (int)Math.Ceiling(TotalCount / (double)PageSize); } }
        public int TotalCount { get; set; }
        public IList<T> Models { get; set; }
    }
}
